gapis/api/vulkan/vulkan.api

```diff
@@ -8700,6 +8700,229 @@ class VkDebugMarkerMarkerInfoEXT {
+	  	ShaderModules[as!VkShaderModule(obj)].DebugInfo = updateDebugMarkerObjectName(
+          ShaderModules[as!VkShaderModule(obj)].DebugInfo, nameInfo)
+	  case VK_DEBUG_REPORT_OBJECT_TYPE_PIPELINE_CACHE_EXT:
+	  	PipelineCaches[as!VkPipelineCache(obj)].DebugInfo = updateDebugMarkerObjectName(
+          PipelineCaches[as!VkPipelineCache(obj)].DebugInfo, nameInfo)
+	  case VK_DEBUG_REPORT_OBJECT_TYPE_PIPELINE_LAYOUT_EXT:
+	  	PipelineLayouts[as!VkPipelineLayout(obj)].DebugInfo = updateDebugMarkerObjectName(
+          PipelineLayouts[as!VkPipelineLayout(obj)].DebugInfo, nameInfo)
+	  case VK_DEBUG_REPORT_OBJECT_TYPE_RENDER_PASS_EXT:
+	  	RenderPasses[as!VkRenderPass(obj)].DebugInfo = updateDebugMarkerObjectName(
+          RenderPasses[as!VkRenderPass(obj)].DebugInfo, nameInfo)
+	  case VK_DEBUG_REPORT_OBJECT_TYPE_PIPELINE_EXT:
+      if (as!VkPipeline(obj) in GraphicsPipelines) {
+        GraphicsPipelines[as!VkPipeline(obj)].DebugInfo = updateDebugMarkerObjectName(
+            GraphicsPipelines[as!VkPipeline(obj)].DebugInfo, nameInfo)
+      } else if (as!VkPipeline(obj) in ComputePipelines) {
+        ComputePipelines[as!VkPipeline(obj)].DebugInfo = updateDebugMarkerObjectName(
+            ComputePipelines[as!VkPipeline(obj)].DebugInfo, nameInfo)
+      }
+	  case VK_DEBUG_REPORT_OBJECT_TYPE_DESCRIPTOR_SET_LAYOUT_EXT:
+	  	DescriptorSetLayouts[as!VkDescriptorSetLayout(obj)].DebugInfo = updateDebugMarkerObjectName(
+        DescriptorSetLayouts[as!VkDescriptorSetLayout(obj)].DebugInfo, nameInfo)
+	  case VK_DEBUG_REPORT_OBJECT_TYPE_SAMPLER_EXT:
+	  	Samplers[as!VkSampler(obj)].DebugInfo = updateDebugMarkerObjectName(
+        Samplers[as!VkSampler(obj)].DebugInfo, nameInfo)
+	  case VK_DEBUG_REPORT_OBJECT_TYPE_DESCRIPTOR_POOL_EXT:
+	  	DescriptorPools[as!VkDescriptorPool(obj)].DebugInfo = updateDebugMarkerObjectName(
+        DescriptorPools[as!VkDescriptorPool(obj)].DebugInfo, nameInfo)
+	  case VK_DEBUG_REPORT_OBJECT_TYPE_DESCRIPTOR_SET_EXT:
+	  	DescriptorSets[as!VkDescriptorSet(obj)].DebugInfo = updateDebugMarkerObjectName(
+        DescriptorSets[as!VkDescriptorSet(obj)].DebugInfo, nameInfo)
+	  case VK_DEBUG_REPORT_OBJECT_TYPE_FRAMEBUFFER_EXT:
+	  	Framebuffers[as!VkFramebuffer(obj)].DebugInfo = updateDebugMarkerObjectName(
+        Framebuffers[as!VkFramebuffer(obj)].DebugInfo, nameInfo)
+	  case VK_DEBUG_REPORT_OBJECT_TYPE_COMMAND_POOL_EXT:
+	  	CommandPools[as!VkCommandPool(obj)].DebugInfo = updateDebugMarkerObjectName(
+        CommandPools[as!VkCommandPool(obj)].DebugInfo, nameInfo)
+	  case VK_DEBUG_REPORT_OBJECT_TYPE_SURFACE_KHR_EXT:
+	  	Surfaces[as!VkSurfaceKHR(obj)].DebugInfo = updateDebugMarkerObjectName(
+          Surfaces[as!VkSurfaceKHR(obj)].DebugInfo, nameInfo)
+	  case VK_DEBUG_REPORT_OBJECT_TYPE_SWAPCHAIN_KHR_EXT:
+	  	Swapchains[as!VkSwapchainKHR(obj)].DebugInfo = updateDebugMarkerObjectName(
+          Swapchains[as!VkSwapchainKHR(obj)].DebugInfo, nameInfo)
+  }
+}
+
+@override
+@no_replay
```

Comment:
Will be changed later to 'custom' once we start to use the debug markers for grouping